---
description: Compilerwarnung (Stufe 1) CS1058
title: Compilerwarnung (Stufe 1) CS1058
ms.date: 07/20/2015
f1_keywords:
- CS1058
helpviewer_keywords:
- CS1058
ms.assetid: ed50590c-f130-47c3-976d-322a6c8f996d
ms.openlocfilehash: 77097472f95d636a6ebc79a5a86083042a48ca32
ms.sourcegitcommit: d579fb5e4b46745fd0f1f8874c94c6469ce58604
ms.translationtype: HT
ms.contentlocale: de-DE
ms.lasthandoff: 08/30/2020
ms.locfileid: "89139124"
---
# <a name="compiler-warning-level-1-cs1058"></a><span data-ttu-id="2a5aa-103">Compilerwarnung (Stufe 1) CS1058</span><span class="sxs-lookup"><span data-stu-id="2a5aa-103">Compiler Warning (level 1) CS1058</span></span>
<span data-ttu-id="2a5aa-104">Eine vorangehende catch-Klausel fängt bereits alle Ausnahmen dieses Typs ab.</span><span class="sxs-lookup"><span data-stu-id="2a5aa-104">A previous catch clause already catches all exceptions.</span></span> <span data-ttu-id="2a5aa-105">Alle ausgelösten Ausnahmen werden in folgender Ausnahme umschlossen: System.Runtime.CompilerServices.RuntimeWrappedException</span><span class="sxs-lookup"><span data-stu-id="2a5aa-105">All exceptions thrown will be wrapped in a System.Runtime.CompilerServices.RuntimeWrappedException</span></span>  
  
 <span data-ttu-id="2a5aa-106">Dieses Attribut löst CS1058 aus, wenn ein `catch()`-Block nach einem `catch (System.Exception e)`-Block keinen bestimmten Ausnahmetyp aufweist.</span><span class="sxs-lookup"><span data-stu-id="2a5aa-106">This attribute causes CS1058 if a `catch()` block has no specified exception type after a `catch (System.Exception e)` block.</span></span> <span data-ttu-id="2a5aa-107">Die Warnung schlägt vor, dass der `catch()`-Block keine Ausnahmen erfassen soll.</span><span class="sxs-lookup"><span data-stu-id="2a5aa-107">The warning advises that the `catch()` block will not catch any exceptions.</span></span>  
  
<span data-ttu-id="2a5aa-108">Ein `catch()`-Block nach einem `catch (System.Exception e)`-Block kann Nicht-CLS-Ausnahmen erfassen, wenn `RuntimeCompatibilityAttribute` in der Datei „AssemblyInfo.cs“ auf FALSE festgelegt ist: `[assembly: RuntimeCompatibilityAttribute(WrapNonExceptionThrows = false)]`.</span><span class="sxs-lookup"><span data-stu-id="2a5aa-108">A `catch()` block after a `catch (System.Exception e)` block can catch non-CLS exceptions if the `RuntimeCompatibilityAttribute` is set to false in the AssemblyInfo.cs file: `[assembly: RuntimeCompatibilityAttribute(WrapNonExceptionThrows = false)]`.</span></span> <span data-ttu-id="2a5aa-109">Wenn dieses Attribut nicht explizit auf FALSE festgelegt ist, werden alle ausgelösten Nicht-CLS-Ausnahmen als Ausnahmen umschlossen und vom `catch (System.Exception e)`-Block erfasst.</span><span class="sxs-lookup"><span data-stu-id="2a5aa-109">If this attribute is not set explicitly to false, all thrown non-CLS exceptions are wrapped as Exceptions and the `catch (System.Exception e)` block catches them.</span></span> <span data-ttu-id="2a5aa-110">Weitere Informationen finden Sie unter [Vorgehensweise: Abfangen einer Nicht-CLS-Ausnahme](../../programming-guide/exceptions/how-to-catch-a-non-cls-exception.md).</span><span class="sxs-lookup"><span data-stu-id="2a5aa-110">For more information, see [How to catch a non-CLS exception](../../programming-guide/exceptions/how-to-catch-a-non-cls-exception.md).</span></span>
  
## <a name="example"></a><span data-ttu-id="2a5aa-111">Beispiel</span><span class="sxs-lookup"><span data-stu-id="2a5aa-111">Example</span></span>  
 <span data-ttu-id="2a5aa-112">Im folgenden Beispiel wird CS1058 generiert.</span><span class="sxs-lookup"><span data-stu-id="2a5aa-112">The following example generates CS1058.</span></span>  
  
```csharp  
// CS1058.cs  
// CS1058 expected  
using System.Runtime.CompilerServices;  
  
// the following attribute is set to true by default in the C# compiler  
// set to false in your source code to resolve CS1058  
[assembly: RuntimeCompatibilityAttribute(WrapNonExceptionThrows = true)]  
  
class TestClass
{  
   static void Main()
   {  
      try {}  
  
      catch (System.Exception e) {
         System. Console.WriteLine("Caught exception {0}", e);  
      }  
  
      catch {}   // CS1058. This line will never be reached.  
   }  
}  
```
